sphinx/ext/autosummary/generate.py

```diff
@@ -393,6 +394,14 @@ def generate_autosummary_docs(sources: List[str], output_dir: str = None,
     # keep track of new files
     new_files = []
 
+    if app:
+        filename_map = app.config.autosummary_filename_map
+        if not isinstance(filename_map, Mapping):
+            raise TypeError('autosummary_filename_map should be a mapping from '
+                            'strings to strings')
```

Comment:
You don't need to check the value is dict here. The Sphinx automatically checks the given value is the same type as the default value. But it does not check the contents of the dict. I think the default check is enough.

Note: If you'd like to add more careful validations, please use a hook for the config-inited event to validate user configurations.

https://github.com/sphinx-doc/sphinx/blob/98a854deca55dc14b66e222fd94dcecc993e691d/sphinx/builders/latex/__init__.py#L498-L503
https://github.com/sphinx-doc/sphinx/blob/98a854deca55dc14b66e222fd94dcecc993e691d/sphinx/builders/latex/__init__.py#L558